src/Symfony/Component/Finder/Comparator/NumberComparator.php

```diff
@@ -41,8 +41,8 @@ class NumberComparator extends Comparator
      */
     public function __construct(?string $test)
     {
-        if (!preg_match('#^\s*(==|!=|[<>]=?)?\s*([0-9\.]+)\s*([kmg]i?)?\s*$#i', $test, $matches)) {
-            throw new \InvalidArgumentException(sprintf('Don\'t understand "%s" as a number test.', $test));
+        if (null === $test || !preg_match('#^\s*(==|!=|[<>]=?)?\s*([0-9\.]+)\s*([kmg]i?)?\s*$#i', $test, $matches)) {
+            throw new \InvalidArgumentException(sprintf('Don\'t understand "%s" as a number test.', $test ?? 'NULL'));
```

Comment:
this changed the exception message, right? why not, but the I'd suggest making `null` lowercase

Reply:
Yes, I found it more helpful to print `null` instead of `""`, if `null` is a possible value here. I agree about making it lowercase.